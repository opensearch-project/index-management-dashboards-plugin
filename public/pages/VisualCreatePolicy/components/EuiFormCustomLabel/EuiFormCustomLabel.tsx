/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

import React from "react";
import { EuiText } from "@elastic/eui";

interface EuiFormCustomLabelProps {
  title: string;
  helpText?: string;
  learnMore?: JSX.Element | null | undefined;
  textStyle?: object;
  headerStyle?: object;
  isInvalid?: boolean;
}

// New pattern for label and help text both being above the form row instead of label above and help below.
const EuiFormCustomLabel = ({
  title,
  helpText,
  learnMore = null,
  textStyle = { marginBottom: "5px" },
  headerStyle = { marginBottom: "2px" },
  isInvalid = false,
}: EuiFormCustomLabelProps) => (
  <EuiText style={textStyle}>
    <h5 style={headerStyle} className={`euiFormLabel ${isInvalid ? "euiFormLabel-isInvalid" : ""}`}>
      {title}
    </h5>
    <p>
      {" "}
      {/* Keep the <p> tag even if no helpText to remove last child styling on h tags */}
      {helpText && (
        <span style={{ fontWeight: 200, fontSize: "12px" }}>
          {helpText} {learnMore}
        </span>
      )}
    </p>
  </EuiText>
);

export default EuiFormCustomLabel;
