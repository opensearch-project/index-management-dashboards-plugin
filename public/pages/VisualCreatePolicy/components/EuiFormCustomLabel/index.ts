/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import EuiFormCustomLabel from "./EuiFormCustomLabel";

export default EuiFormCustomLabel;
