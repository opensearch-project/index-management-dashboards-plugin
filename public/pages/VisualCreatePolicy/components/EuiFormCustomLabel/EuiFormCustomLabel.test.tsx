/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { render } from "@testing-library/react";
import EuiFormCustomLabel from "./EuiFormCustomLabel";

describe("<EuiFormCustomLabel /> spec", () => {
  it("renders the component", () => {
    const { container } = render(<EuiFormCustomLabel title="Some title" helpText="Some helpful text" />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
