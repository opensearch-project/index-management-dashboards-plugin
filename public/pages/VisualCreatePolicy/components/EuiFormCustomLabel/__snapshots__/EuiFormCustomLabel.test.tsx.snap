// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EuiFormCustomLabel /> spec renders the component 1`] = `
<div
  class="euiText euiText--medium"
  style="margin-bottom: 5px;"
>
  <h5
    class="euiFormLabel "
    style="margin-bottom: 2px;"
  >
    Some title
  </h5>
  <p>
     
    <span
      style="font-weight: 200; font-size: 12px;"
    >
      Some helpful text
       
    </span>
  </p>
</div>
`;
